cumulusci/tasks/bulkdata/load.py

```diff
@@ -120,7 +116,70 @@ def _init_options(self, kwargs):
             self.options.get("set_recently_viewed", True)
         )
 
+    def _init_dataset(self):
+        org_shape_match_only = process_bool_arg(
+            self.options.get("org_shape_match_only", False)
+        )
+        if org_shape_match_only:
+            self.options["mapping"] = None
+            self.options["sql_path"] = None
+            self.options["database_url"] = None
+
+        self.options.setdefault("database_url", None)
+        if self.options.get("database_url"):
+            # prefer database_url if it's set
+            self.options["sql_path"] = None
+        elif self.options.get("sql_path"):
+            self.options.setdefault("mapping", "datasets/mapping.yml")
+        elif self.options.get("mapping"):
+            self.options.setdefault("sql_path", "datasets/sample.sql")
+        elif found_dataset := (
+            self._find_matching_dataset()
+            or (self._find_default_dataset() if not org_shape_match_only else None)
```

Comment:
How about a walrus _or_ nested boolean logic, but not both? This is hard to follow.